### PR TITLE
[macros,numerics] Add and use numbered 'formula' environment

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes ISO/CS comment (C++23 proof)

- [ ] This is a formula which should be labelled Formula (1).  See  HYPERLINK "https://www.iso.org/sites/directives/current/part2/index.xhtml#_idTextAnchor410"ISO/IEC Directives Part 2, 2021, 27.4  "Please number mathematical formulae in the document, for cross-referencing purposes. Arabic numbers in parentheses shall be used, starting with 1. If a mathematical formula is numbered, it should also be referred to in the text and its purpose should be made clear by its context, for example, with an introductory proposition, e.g.  "see 10.1, Formula (3)".
- [ ] see comment above regarding Formulae
- [ ] this appears to be a formulae - please follow the ISO/IEC Directives Part 2, Clause 27

### CODE REVIEW

**Comment by tkoeppe** [comment]
(Have you got a screenshot by any chance? :-) )

**Comment by jensmaurer** [comment]
image

image

**Comment by tkoeppe** [comment]
Looks great, thanks!

File: source/numerics.tex [hunk 1]
```diff
@@ -9648,13 +9679,16 @@
 
 \pnum
 \returns
-\[ \mathsf{L}_n^m(x) =
-   (-1)^m \frac{\mathsf{d} ^ m}{\mathsf{d}x ^ m} \, \mathsf{L}_{n+m}(x)
-   \text{ ,\quad for $x \ge 0$,} \]
+The value given by \eqref{sf.cmath.assoc.laguerre},
```
Comment (tkoeppe) [inline source/numerics.tex:9682]: Maybe we need to say more here. The formula isn't a "value" as such. Maybe "the value L_n^m(x), where L_n^m is given by Formula ..."?

Comment (jensmaurer) [reply source/numerics.tex:9682]: Maybe we can omit the "The value given by" and just say the function returns "L_n^m(x)` ..." and then the rest from your suggestion.

Comment (jensmaurer) [reply source/numerics.tex]: Updated; please have a look.

**Comment by jensmaurer** [comment]
image

**Comment by jensmaurer** [comment]
When we do this, we might replace the "k is k" stuff by directly putting in the monospace-font C++ variables into the "value", e.g. "Pi(nu, k)".  What do you think?

**Comment by tkoeppe** [comment]
> When we do this, we might replace the "k is k" stuff by directly putting in the monospace-font C++ variables into the "value", e.g. "Pi(nu, k)". What do you think?

This is something we've been wanting to do for a while, but this requires some thought. Let's do this for the WD only, and talk about it separately.

@jwakely ^^^ I think you had some thoughts here, too.

**Review by tkoeppe — APPROVED** [review]
This is great, thanks!

**Comment by jwakely** [comment]
> @jwakely ^^^ I think you had some thoughts here, too.

More than just thoughts, see #4295

**Pull request merged**